File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse121.java

public class Lse121 {
    public static void demonstrateMemoryCopy() {
        // Allocate two chunks of memory (using byte arrays to simulate memory chunks)
        byte[] chunk1 = new byte[10];
        byte[] chunk2 = new byte[10];
        
        // Fill first chunk with some data
        for (int i = 0; i < chunk1.length; i++) {
            chunk1[i] = (byte) (i + 65); // ASCII 'A' onwards
        }
        
        // Copy contents of first chunk into second chunk
        System.arraycopy(chunk1, 0, chunk2, 0, chunk1.length);
        
        // Print contents of second chunk
        System.out.println("Contents of chunk2 after copy:");
        for (byte b : chunk2) {
            System.out.print((char) b + " ");
        }
        System.out.println();
        
        // Free memory allocated to first chunk (set to null for garbage collection)
        chunk1 = null;
        System.gc(); // Suggest garbage collection
        
        // Print contents of second chunk again
        System.out.println("Contents of chunk2 after freeing chunk1:");
        for (byte b : chunk2) {
            System.out.print((char) b + " ");
        }
        System.out.println();
        
        // Free memory allocated to second chunk
        chunk2 = null;
        System.gc(); // Suggest garbage collection
        
        System.out.println("Both chunks freed.");
    }
    
    public static void main(String[] args) {
        System.out.println("Test Case 1:");
        demonstrateMemoryCopy();
        
        System.out.println("\\nTest Case 2:");
        demonstrateMemoryCopy();
        
        System.out.println("\\nTest Case 3:");
        demonstrateMemoryCopy();
        
        System.out.println("\\nTest Case 4:");
        demonstrateMemoryCopy();
        
        System.out.println("\\nTest Case 5:");
        demonstrateMemoryCopy();
    }
}

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse121.cpp

#include <iostream>
#include <cstring>
#include <cstdlib>

class Lse121 {
public:
    static void demonstrateMemoryCopy() {
        // Allocate two chunks of memory
        int size = 10;
        char* chunk1 = (char*)malloc(size * sizeof(char));
        char* chunk2 = (char*)malloc(size * sizeof(char));
        
        // Fill first chunk with some data
        for (int i = 0; i < size; i++) {
            chunk1[i] = 'A' + i;
        }
        
        // Copy contents of first chunk into second chunk
        memcpy(chunk2, chunk1, size);
        
        // Print contents of second chunk
        std::cout << "Contents of chunk2 after copy:" << std::endl;
        for (int i = 0; i < size; i++) {
            std::cout << chunk2[i] << " ";
        }
        std::cout << std::endl;
        
        // Free memory allocated to first chunk
        free(chunk1);
        chunk1 = nullptr;
        
        // Print contents of second chunk again
        std::cout << "Contents of chunk2 after freeing chunk1:" << std::endl;
        for (int i = 0; i < size; i++) {
            std::cout << chunk2[i] << " ";
        }
        std::cout << std::endl;
        
        // Free memory allocated to second chunk
        free(chunk2);
        chunk2 = nullptr;
        
        std::cout << "Both chunks freed." << std::endl;
    }
};

int main() {
    std::cout << "Test Case 1:" << std::endl;
    Lse121::demonstrateMemoryCopy();
    
    std::cout << "\\nTest Case 2:" << std::endl;
    Lse121::demonstrateMemoryCopy();
    
    std::cout << "\\nTest Case 3:" << std::endl;
    Lse121::demonstrateMemoryCopy();
    
    std::cout << "\\nTest Case 4:" << std::endl;
    Lse121::demonstrateMemoryCopy();
    
    std::cout << "\\nTest Case 5:" << std::endl;
    Lse121::demonstrateMemoryCopy();
    
    return 0;
}
